### PR TITLE
fix(*): Fixes various Clippy issues and adds an Action pipeline

### LINKED ISSUE #10: Setup actions pipelines
We are currently missing automated testing in the project. Add a GH action pipeline for automated testing

### PR DESCRIPTION
**Pull request by thomastaylor312**
Closes #10

### CODE REVIEW

**Comment by thomastaylor312** [comment]
The pipeline is running here so we can check if it is running before merge: https://github.com/thomastaylor312/krustlet/actions/runs/48304726

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: Build
+        run: |
+          # we need a working Go environment in order to build the Go static library
+          export GOPATH=$HOME/go && export GOBIN=$(go env GOPATH)/bin && export PATH=$PATH:$GOPATH && export PATH=$PATH:$GOBIN
+          mkdir -p $GOPATH/pkg && mkdir -p $GOBIN && mkdir -p $GOPATH/src/github.com/$GITHUB_REPOSITORY
+          mv $(pwd)/* $GOPATH/src/github.com/$GITHUB_REPOSITORY && cd $GOPATH/src/github.com/$GITHUB_REPOSITORY
+          go get -u github.com/golang/dep/cmd/dep
```
Comment (radu-matei) [inline .github/workflows/build.yml]: TODO (not in this PR): update OCI library to use Go modules.

Comment (thomastaylor312) [reply .github/workflows/build.yml]: I had removed these and forgot to save 😆

**Comment by thomastaylor312** [comment]
Passing build: https://github.com/thomastaylor312/krustlet/runs/480271596?check_suite_focus=true

**Comment by bketelsen** [comment]
LGTM

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: Build
+        run: |
+          cargo fmt --all -- --check
+          cargo install just
```
Comment (rylev) [inline .github/workflows/build.yml]: I don't think `just` is worth it here. Can we not use it for CI?

File: src/kubelet.rs [hunk 3]
```diff
@@ -320,7 +320,7 @@ pub trait Provider {
                             if let Some(cfkey) = env_src.field_ref {
                                 return fields
                                     .get(cfkey.field_path.as_str())
-                                    .and_then(|s| Some(s.to_string()))
+                                    .map(|s| s.to_string())
```
Comment (rylev) [inline src/kubelet.rs]: I know it was `to_string` before, but can we turn this into `to_owned`?

Comment (thomastaylor312) [reply src/kubelet.rs]: Oops, I meant to do that and forgot

**Pull request merged**